models/location_helpers.go

```diff
@@ -274,6 +278,75 @@ func ChangeBackupLocation(q *reform.Querier, locationID string, params ChangeBac
 	return row, nil
 }
 
+func testS3Config(c *S3LocationConfig) error {
+	parsedURL, err := url.Parse(c.Endpoint)
+	if err != nil {
+		return err
+	}
+
+	endpoint := path.Join(parsedURL.Host, parsedURL.Path)
+	secure := true
+	if parsedURL.Scheme == "http" {
+		secure = false
+	}
+	minioClient, err := minio.New(endpoint, c.AccessKey, c.SecretKey, secure)
+	if err != nil {
+		return err
+	}
+
+	exists, err := minioClient.BucketExists(c.BucketName)
```

Comment:
Is it enough to check if bucket exists? This tests only some read scope (not sure which one).

Reply:
As I understand there is no easy way to understand what permissions do you have. There are different mechanisms to grant access in different storage implementations.

Reply:
@artemgavrilov yup